api/envoy/extensions/filters/http/file_system_buffer/v3/file_system_buffer.proto

```diff
@@ -0,0 +1,114 @@
+
+    // Never buffer, do nothing.
+    bool bypass = 2 [(validate.rules).bool = {const: true}];
+
+    // If ``content-length`` is not present, buffer the entire input,
+    // inject ``content-length`` header, then output.
+    // If ``content-length`` is already present, act like ``stream_when_possible``.
+    bool inject_content_length_if_necessary = 3 [(validate.rules).bool = {const: true}];
+
+    // Always buffer the entire input, and inject ``content-length``,
+    // overwriting any provided content-length header.
+    bool fully_buffer_and_always_inject_content_length = 4 [(validate.rules).bool = {const: true}];
+
+    // Always buffer the entire input, do not inject ``content-length``.
+    bool fully_buffer = 5 [(validate.rules).bool = {const: true}];
+  }
+}
+
+// [#not-implemented-hide:]
+// The configuration for one direction of the filter behavior.
+message StreamConfig {
+  // Whether to bypass / stream / fully buffer / etc.
+  // May be unset, in which case the default is ``stream_when_possible``.
+  BufferBehavior behavior = 1;
+
+  // The amount stored in the memory buffer before buffering to disk.
+  google.protobuf.UInt64Value memory_buffer_bytes_limit = 2;
+
+  // The maximum storage (excluding memory) to be buffered in this filter.
+  google.protobuf.UInt64Value storage_buffer_bytes_limit = 3;
+
+  // The maximum amount that can be queued for writing to storage, above which the
+  // source is requested to pause. If unset, defaults to the same value as
+  // ``memory_buffer_bytes_limit``.
+  //
+  // For example, assuming the recipient is not consuming data at all, if
+  // ``memory_buffer_bytes_limit`` was 32*1024, and ``storage_buffer_queue_high_watermark_bytes``
+  // was 64*1024, and the filesystem is backed up so writes are not occurring promptly,
+  // then:
+  // * Any request less than 32*1024 will eventually pass through without ever attempting
+  //   to write to disk.
+  // * Any request with over 32*1024 buffered will start trying to write to disk.
+  //   * If it reaches (32+64)*1024 buffered and not yet written to disk, a high
+  //     watermark signal is sent to the source.
+  // * Any stream whose total size exceeds
+  //   ``memory_buffer_bytes_limit + storage_buffer_bytes_limit`` will provoke an error.
+  //   (Note, if the recipient *is* consuming data then it is possible for such an
+  //   oversized request to pass through the buffer filter, provided the recipient
+  //   isn't consuming data too slowly.)
+  //
+  // The low watermark signal is sent when the memory buffer is at size
+  //  ``memory_buffer_bytes_limit + (storage_buffer_bytes_limit / 2)``.
+  google.protobuf.UInt64Value storage_buffer_queue_high_watermark_bytes = 4;
+}
+
+// [#not-implemented-hide:]
+// Configuration for file system buffer filters.
+//
+// Most fields are wrapped in well-known types so that we can detect the difference between
+// unset and explicitly zero.
```

Comment:
this generates user facing docs, so I would remove this. This is the standard and we don't call this out elsewhere.